source/ranges.tex

```diff
@@ -5358,17 +5436,23 @@
 \effects
 Equivalent to:
 \begin{codeblock}
-auto cur = x.i_.@\placeholder{current}@;
-auto end = ranges::end(x.i_.parent_->base_);
-if (cur == end) return true;
 auto [pcur, pend] = subrange{x.i_.parent_->pattern_};
-if (pcur == pend) return x.incremented_;
-do {
+auto end = ranges::end(x.i_.parent_->base_);
+if constexpr (@\exposconcept{tiny-range}@<Pattern>) {
+  const auto & cur = x.i_.current ;
+  if (cur == end) return true;
+  if (pcur == pend) return x.incremented_;
+  return *cur == *pcur;
+} else {
+  auto cur = x.i_.current ;
+  if (cur == end) return true;
+  if (pcur == pend) return x.incremented_;
+  do {
   if (*cur != *pcur) return false;
   if (++pcur == pend) return true;
 } while (++cur != end);
```

Comment:
The closing brace of the do while needs to be align with the `do`, and its statements indented two more spaces.